balena/models/device.py

```diff
@@ -930,8 +932,8 @@ def move(self, uuid, app_name):
             raise exceptions.ApplicationNotFound(app_name)
         application = application[0]
 
-        device_dev_type = list(filter(lambda dev_type: dev_type['slug'] == device['is_of__device_type'][0]['slug'], self.config.get_device_types()))[0]
-        app_dev_type = list(filter(lambda dev_type: dev_type['slug'] == application['is_for__device_type'][0]['slug'], self.config.get_device_types()))[0]
+        device_dev_type = list(filter(lambda dev_type: dev_type['slug'] == device['is_of__device_type'][0]['slug'], self.device_type.get_all_supported()))[0]
```

Comment:
Could we fetch the `self.device_type.get_all_supported()` only once before the two filters?